### ISSUE TITLE
Accessibility issues: Several form components cannot be accessed using browse mode

### ISSUE THREAD

**Issue by brunoprietog** [uber/baseweb#4121]
## Current Behavior

Using the NVDA screen reader in browse mode, when I encounter items such as a combobox or menu, I cannot enter them using space or enter. The only way to access them is to return to a previous control and press tab.

## Expected Behavior

When navigating in browse mode with NVDA, using the arrows, if I encounter any of these items, I should be able to access them using space or enter, just as I can do with a Select component for example.

## Your Environment

It can be tested on the same page of the documentation. I also tested it in my environment and the same thing happens.

- [There are problems here](https://baseweb.design/components/combobox/)
- [There are problems here](https://baseweb.design/components/menu/)
- [Here it works well](https://baseweb.design/components/select/)

| Tech    | Version |
| ------- | ------- |
| Base UI | v9.111.2  |
| React   |      17.0.1   |
| browser |     Microsoft Edge Chromium    |

- [x] I have searched the [issues](https://github.com/uber/baseweb/issues) of this repository and believe that this is not a duplicate.

**Comment by brunoprietog** [comment]
Same behavior [here](https://baseweb.design/components/radio/), with Radio

**Comment by chasestarr** [comment]
Thank you for reporting. We will address this soon

**Comment by chasestarr** [comment]
Thank you for reporting these issues. I've done some fixes on combobox already and have been trying to find solutions for menu/listbox and radio/radiogroup.

When inspecting the W3C aria guidelines for radio groups, [this example](https://www.w3.org/TR/wai-aria-practices-1.2/examples/listbox/listbox-scrollable.html) also cannot enter the control with enter. Similarly for W3C listbox guidelines, I'm finding the same behavior on [this example](https://www.w3.org/TR/wai-aria-practices-1.2/examples/listbox/listbox-scrollable.html).

Do you mind pointing to some examples that operate as-expected?

**Comment by brunoprietog** [comment]
Thanks for the work. [This combobox works fine](https://www.w3.org/TR/wai-aria-practices-1.2/examples/combobox/combobox-autocomplete-both.html), [this radiogroup also works fine with tabindex](https://www.w3.org/TR/wai-aria-practices-1.2/examples/radio/radio-1/radio-1.html), but unfortunately the listbox does not. I suspect it's a bug in NVDA that it can't get the focus from a listbox, since any item that has another role and tabindex=0 does work, except listbox. I will report this in the repository. However, in Microsoft's Fluent UI it works fine. Apparently they are using data-is-focusable and they handle the focus, because they use tabindex at -1. [Here is the example](https://fluentsite.z22.web.core.windows.net/0.53.0/components/list/definition) I am referring to (Selectable list).

**Comment by brunoprietog** [comment]
[Here is the issue](https://github.com/nvaccess/nvda/issues/12266) I opened in the NVDA repository so you can follow up on it. Anyway it would be interesting if you can review the approach they took to it in Fluent UI to see if there is a possibility to fix it from here.
Thank you very much!
